@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1627,17 +1638,19 @@ export class TextElementsRenderer {
                 this.m_screenCollisions
             );
             if (result === PlacementResult.Invisible) {
+                // Reset the icon and keep it from popping up before fading in.
+                iconInvisible = true;
                 iconRenderState.reset();
-
-                if (placementStats) {
+                iconRenderState.opacity = 0;
+                if (placementStats !== undefined) {
                     ++placementStats.numNotVisible;
                 }
-                return false;
+                // Proceed with the label text, which may still be rendered.
```

Comment:
Please add a unit test in TextElementsRendererTest to cover this change.

Reply:
outdated